examples/generate_image_dataset/generate_images.py

```diff
@@ -74,34 +74,41 @@ def single_step(self, action=None, get_observation=False):
     # load config
     config = spear.get_config(user_config_files=[os.path.realpath(os.path.join(os.path.dirname(__file__), "user_config.yaml"))])
 
+    # configure system based on config
+    spear.configure_system(config)
+
     # read data from csv
     df = pd.read_csv(args.poses_file)
 
-    # create dir for storing images
-    if not args.benchmark:
-        for render_pass in config.SIMULATION_CONTROLLER.CAMERA_AGENT.CAMERA.RENDER_PASSES:
-            render_pass_dir = os.path.realpath(os.path.join(args.images_dir, render_pass))
-            shutil.rmtree(render_pass_dir, ignore_errors=True)
-            os.makedirs(render_pass_dir)
+    # create SpEngine object
+    sp_instance = spear.Instance(config)
+
+    # create Env object
+    env = CustomEnv(config, sp_instance, num_internal_steps=args.num_internal_steps)
 
     # iterate over all poses
     prev_scene_id = ""
+
     for pose in df.to_records():
 
         # if the scene_id of our current pose has changed, then create a new Env
         if pose["scene_id"] != prev_scene_id:
 
+            # create dir for storing images
+            if not args.benchmark:
+                for render_pass in config.SP_ENGINE.LEGACY.CAMERA_AGENT.CAMERA.RENDER_PASSES:
+                    render_pass_dir = os.path.realpath(os.path.join(args.images_dir, pose["scene_id"], render_pass))
+                    shutil.rmtree(render_pass_dir, ignore_errors=True)
+                    os.makedirs(render_pass_dir)
+
             # close the previous Env
-            if prev_scene_id != "":
-                env.close()
+            env.close()
 
-            # update scene_id
-            config.defrost()
-            config.SIMULATION_CONTROLLER.SCENE_ID = pose["scene_id"]
-            config.freeze()
+            # open the desired level
+            spear.open_level(sp_instance, pose["scene_id"])
```

Comment:
`open_level(scene_id)`, with all of its custom logic to turn `scene_id` into a well-formed path that Unreal understands, should be a helper function in `Env`, it shouldn't be in the global namespace.